Program.cs

```diff
@@ -1,10 +1,133 @@
-﻿namespace Exercises
+﻿using System.Collections;
+using System.Collections.Generic;
+using static Exercises.Program;
+
+namespace Exercises
 {
-    internal class Program
+    public static class Program
     {
-        static void Main(string[] args)
+        private static void Main()
         {
-            Console.WriteLine("Hello, World!");
+            //Exercise 1;
+
+            var myList = new List<int> { 5543210, 273, 8, 41, 87643, 6 };
+            IEnumerable<int> evenNumbers;
+
+            GetEvenNumbers(myList);
+
+            foreach (var number in evenNumbers)
+            {
+                Console.Write(number + " ");
+            }
+
+            void GetEvenNumbers(List<int> myList)
+            {
+                evenNumbers = myList.Where(n => n % 2 == 0);
+            }
+
+            //Exercise 2;
+
+            List<Student> students = new List<Student>()
+            {
+                new Student { StudentName = "John", Age = 20 },
+                new Student { StudentName = "Alice", Age = 21 },
+                new Student { StudentName = "Bob", Age = 22 },
+                new Student { StudentName = "Kate", Age = 20 },
+                new Student { StudentName = "Mike", Age = 21 }
+            };
+
+            IEnumerable<IGrouping<int, Student>> groupedStudents;
+
+            GetGroupStudents(students);
+
+            foreach (var group in groupedStudents)
+            {
+                Console.WriteLine($"Age: {group.Key}, Count: {group.Count()}");
+            }
+
+            void GetGroupStudents(IEnumerable<Student> students)
+            {
+                groupedStudents = from student in students
```

Comment:
This is an interesting option, but is not preferred, because. difficult to understand in C#
Use similar LINQ methods, they allow you to do the same